openshift/cluster-network-operator
# Bug 1989246: use new default leader election values to handle apiserver rollout on SNO

In case of CNO gets deployed on SNO, it can run into leader election timeouts during an api server rollout (as only one api server is available and it doesn't restart fast enough yet). 
Therefor the default values for leader election have been updated (https://github.com/openshift/library-go/pull/1104) to the [recommended](https://github.com/openshift/enhancements/blame/84e894ead7b188a1013556e0ba6973b8463995f1/CONVENTIONS.md#L183) values.

This PR updates the version of the openshift/library-go library to use these new values and being able to handle a 60 seconds downtime of the api server on SNO.

**How to reproduce / validate:**
1. Start a SNO cluster with this patch (e.g. via clusterbot `launch 4.9.0-0.nightly,openshift/cluster-network-operator#1175 aws,single-node`)
2. Trigger a apiserver rollout (e.g. via `k patch kubeapiserver/cluster --type merge -p '{"spec":{"forceRedeploymentReason":"something"}}'`)
3. Check the logs from CNO controller pod during the update
4. Validate the pod did not get restarted cause it lost the lease (e.g. check no previous logs are available / doesn't contain `leaderelection lost`

**Special notes for reviewers:**
The important file is vendor/github.com/openshift/library-go/pkg/config/leaderelection/leaderelection.go, which uses the new recommended values:

https://github.com/openshift/cluster-network-operator/blob/048145f61ff4d93af0221d079868bbccebe0df2e/vendor/github.com/openshift/library-go/pkg/config/leaderelection/leaderelection.go#L96-L109
